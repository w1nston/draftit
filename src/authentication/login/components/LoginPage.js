import React from 'react';

const LoginPage = () => <form />;

export default LoginPage;
