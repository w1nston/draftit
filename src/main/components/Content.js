import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import LoginPage from '../../authentication/login/components/LoginPage';

export const loginRouteRenderer = isAuthenticated => () => {
  if (isAuthenticated()) {
    return <Redirect to={{ pathname: '/' }} />;
  }
  return <LoginPage />;
};

const Content = ({ isAuthenticated }) => (
  <Switch>
    <Route exact path="/login" render={loginRouteRenderer} />
  </Switch>
);

export default Content;
