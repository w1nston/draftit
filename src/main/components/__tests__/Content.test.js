import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Redirect } from 'react-router-dom';
import Content, { loginRouteRenderer } from '../Content';
import LoginPage from '../../../authentication/login/components/LoginPage';

describe('Content', () => {
  it('renders', () => {
    const component = shallow(<Content />);
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe('loginRouteRenderer', () => {
  it('redirects to root path when authenticated', () => {
    const isAuthenticated = jest.fn(() => true);
    expect(loginRouteRenderer(isAuthenticated)()).toEqual(
      <Redirect to={{ pathname: '/' }} />
    );
  });

  it('renders login page when not authenticated', () => {
    const isAuthenticated = jest.fn(() => false);
    expect(loginRouteRenderer(isAuthenticated)()).toEqual(<LoginPage />);
  });
});
