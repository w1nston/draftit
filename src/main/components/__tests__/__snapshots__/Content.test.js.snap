// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders 1`] = `
<Switch>
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
</Switch>
`;
